src/mame/taito/taito_f3_v.cpp

```diff
@@ -2502,43 +2476,43 @@ inline void taito_f3_state::f3_drawgfx(bitmap_rgb32 &dest_bmp, const rectangle &
 						u32 *dest = dest0;
 						u8 *pri = pri0;
 
-						switch (x)
+						if (x_base > 15)
+						{
+							// pset_o
+							for(int x = x_base & 0xf; x > -1; --x)
+							{
+								const u8 p = *pri;
+								if (!p || p == 0xff)
+								{
+									*dest = pal[*source & m_sprite_pen_mask];
+									*pri = pri_dst;
+								}
+
+								source += dx;
+								dest++;
+								pri++;
+							}
+						}
+						else
 						{
-							int c;
-							u8 p;
-							case 31: PSET_O NEXT_P [[fallthrough]];
-							case 30: PSET_O NEXT_P [[fallthrough]];
-							case 29: PSET_O NEXT_P [[fallthrough]];
-							case 28: PSET_O NEXT_P [[fallthrough]];
-							case 27: PSET_O NEXT_P [[fallthrough]];
-							case 26: PSET_O NEXT_P [[fallthrough]];
-							case 25: PSET_O NEXT_P [[fallthrough]];
-							case 24: PSET_O NEXT_P [[fallthrough]];
-							case 23: PSET_O NEXT_P [[fallthrough]];
-							case 22: PSET_O NEXT_P [[fallthrough]];
-							case 21: PSET_O NEXT_P [[fallthrough]];
-							case 20: PSET_O NEXT_P [[fallthrough]];
-							case 19: PSET_O NEXT_P [[fallthrough]];
-							case 18: PSET_O NEXT_P [[fallthrough]];
-							case 17: PSET_O NEXT_P [[fallthrough]];
-							case 16: PSET_O break;
-
-							case 15: PSET_T NEXT_P [[fallthrough]];
-							case 14: PSET_T NEXT_P [[fallthrough]];
-							case 13: PSET_T NEXT_P [[fallthrough]];
-							case 12: PSET_T NEXT_P [[fallthrough]];
-							case 11: PSET_T NEXT_P [[fallthrough]];
-							case 10: PSET_T NEXT_P [[fallthrough]];
-							case  9: PSET_T NEXT_P [[fallthrough]];
-							case  8: PSET_T NEXT_P [[fallthrough]];
-							case  7: PSET_T NEXT_P [[fallthrough]];
-							case  6: PSET_T NEXT_P [[fallthrough]];
-							case  5: PSET_T NEXT_P [[fallthrough]];
-							case  4: PSET_T NEXT_P [[fallthrough]];
-							case  3: PSET_T NEXT_P [[fallthrough]];
-							case  2: PSET_T NEXT_P [[fallthrough]];
-							case  1: PSET_T NEXT_P [[fallthrough]];
-							case  0: PSET_T
+							//pset_t
+							for(int x = x_base; x > -1; --x)
+							{
```

Comment:
Same comments as for the previous `for` loop – since you don’t actually need `x` in the loop, you can express it as an upward count and avoid the awkward comparison to -1 and dependency on signed numbers.  (Also, the inconsistent spacing.)